File: Java/src/com/matrixmultiply/TimeComparison.java
package com.matrixmultiply;

public class TimeComparison {
  public static void timeComparison(int[][] matrix1, int[][] matrix2) {
    long startTime = System.currentTimeMillis();
    SequentialMultiply.multiply(matrix1, matrix2);
    long endTime = System.currentTimeMillis();
    System.out.println("Sequential time: " + (endTime - startTime) + "ms");

    startTime = System.currentTimeMillis();
    ConcurrentMultiply.multiply(matrix1, matrix2);
    endTime = System.currentTimeMillis();
    System.out.println("Concurrent time: " + (endTime - startTime) + "ms");
  }
}

File: Go/matrix_multiplication/time_comparison.go
package main

import (
	"fmt"
	"time"
)

func TimeComparison(matrix1 [][]int, matrix2 [][]int) {
	startTime := time.Now()
	SequentialMultiply(matrix1, matrix2)
	endTime := time.Now()
	fmt.Println("Sequential time: ", endTime.Sub(startTime).Milliseconds(), "ms")

	startTime = time.Now()
	ConcurrentMultiply(matrix1, matrix2)
	endTime = time.Now()
	fmt.Println("Concurrent time: ", endTime.Sub(startTime).Milliseconds(), "ms")
}
